watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/GLSD_anaerobic_digester/GLSD_anaerobic_digester_ui.py

```diff
@@ -0,0 +1,456 @@
+        description="Water recovery [g-H2O treated/g-H2O inlet]",
+        is_input=True,
+        input_category="Anaerobic digester",
+        is_output=False,
+    )
+    exports.add(
+        obj=fs.AD.reaction_conversion[0, "tss_reaction"],
+        name="TSS conversion",
+        ui_units=pyunits.dimensionless,
+        display_units="fraction",
+        rounding=2,
+        description="TSS conversion [g-TSS reacted/g-TSS inlet]",
+        is_input=True,
+        input_category="Anaerobic digester",
+        is_output=False,
+    )
+    exports.add(
+        obj=fs.AD.generation_ratio["tss_reaction", "methane"],
+        name="CH4 conversion ratio",
+        ui_units=pyunits.dimensionless,
+        display_units="fraction",
+        rounding=4,
+        description="CH4 mass conversion ratio with respect to TSS [g-CH4 produced/g-TSS"
+        " reacted]",
+        is_input=True,
+        input_category="Anaerobic digester",
+        is_output=False,
+    )
+    exports.add(
+        obj=fs.AD.generation_ratio["tss_reaction", "carbon_dioxide"],
+        name="CO2 conversion ratio",
+        ui_units=pyunits.dimensionless,
+        display_units="fraction",
+        rounding=4,
+        description="CO2 mass conversion ratio with respect to TSS [g-CO2 produced/g-TSS"
+        " reacted]",
+        is_input=True,
+        input_category="Anaerobic digester",
+        is_output=False,
+    )
+    exports.add(
+        obj=fs.AD.generation_ratio["tss_reaction", "nitrogen"],
+        name="N2 conversion ratio",
+        ui_units=pyunits.dimensionless,
+        display_units="fraction",
+        rounding=4,
+        description="N2 mass conversion ratio with respect to TSS [g-N2 produced/g-TSS"
+        " reacted]",
+        is_input=True,
+        input_category="Anaerobic digester",
+        is_output=False,
+    )
+    exports.add(
+        obj=fs.AD.generation_ratio["tss_reaction", "oxygen"],
+        name="O2 conversion ratio",
+        ui_units=pyunits.dimensionless,
+        display_units="fraction",
+        rounding=4,
+        description="O2 mass conversion ratio with respect to TSS [g-O2 produced/g-TSS"
+        " reacted]",
+        is_input=True,
+        input_category="Anaerobic digester",
+        is_output=False,
+    )
+    exports.add(
+        obj=fs.AD.HRT,
+        name="HRT",
+        ui_units=pyunits.g / pyunits.m**2 / pyunits.day,
+        display_units="h",
+        rounding=0,
+        description="Hydraulic retention time",
+        is_input=True,
+        input_category="Anaerobic digester",
+        is_output=False,
+    )
+    exports.add(
+        obj=fs.mabr.energy_electric_flow_vol_inlet,
+        name="Blower specific power",
+        ui_units=pyunits.kWh / pyunits.m**3,
+        display_units="kWh/m3",
+        rounding=2,
+        description="Specific power relating the power to the inlet flow",
+        is_input=True,
+        input_category="Anaerobic digester",
+        is_output=False,
+    )
+
```

Comment:
For some units, the pumping (or blower/compressor) energy/costs might be wrapped in rather than adding a pump unit explicitly with its own costing/energy consumption independently.

Reply:
If that's the case, are we double counting when computing the direct_capital_norm (line 297)?